components/script/bluetooth_blacklist.rs

```diff
@@ -0,0 +1,114 @@
+              RefCell::new(BluetoothBlacklist(parse_blacklist())));
+
+macro_rules! return_if_blacklisted {
+    ($uuid:expr, $exclude_fn:ident) => {
+        if BLUETOOTH_BLACKLIST.with(|blist| {
+            blist.borrow().$exclude_fn($uuid.as_ref())
+        }) {
+            return Err(Security);
+        }
+    }
+}
+
+pub struct BluetoothBlacklist(HashMap<String, ExcludeType>);
+
+#[derive(Eq, PartialEq)]
+pub enum ExcludeType {
+    Exclude, // Read and Write
+    ExcludeReads,
+    ExcludeWrites,
+}
+
+impl BluetoothBlacklist {
+
+    // https://webbluetoothcg.github.io/web-bluetooth/#blacklisted
+    pub fn is_excluded(&self, uuid: &str) -> bool {
+        self.0.is_empty() ||
+        self.0.get(uuid).map_or(false, |value| value.eq(&ExcludeType::Exclude))
+    }
+
+    // https://webbluetoothcg.github.io/web-bluetooth/#blacklisted-for-reads
+    pub fn is_excluded_from_reads(&self, uuid: &str) -> bool {
+        self.is_excluded(uuid) ||
+        self.0.get(uuid).map_or(false, |value| value.eq(&ExcludeType::ExcludeReads))
+    }
+
+    // https://webbluetoothcg.github.io/web-bluetooth/#blacklisted-for-writes
+    pub fn is_excluded_from_writes(&self, uuid: &str) -> bool {
+        self.is_excluded(uuid) ||
+        self.0.get(uuid).map_or(false, |value| value.eq(&ExcludeType::ExcludeWrites))
+    }
+}
+
+// https://webbluetoothcg.github.io/web-bluetooth/#parsing-the-blacklist
+fn parse_blacklist() -> HashMap<String, ExcludeType> {
+    // Step 3
+    let mut result = HashMap::new();
+    // Step 1 missing, currently we parse ./resources/gatt_blacklist.txt.(zakorgy)
+    let file = match File::open(FILE_PATH) {
+        Ok(file) => file,
+        Err(_) => return result,
+    };
+    let content = BufReader::new(file);
+    // Step 2 and 4
+    for line in content.lines() {
+        if let Ok(line) = line {
+            // Step 4.1
+            if !line.is_empty() && !line.starts_with('#') {
+                let mut value = ExcludeType::Exclude;
+                let mut words = line.split_whitespace();
+                if let Some(uuid) = words.next() {
+                    // Step 4.3
+                    match words.next() {
+                        Some(EXCLUDE_READS) => {
+                            value = ExcludeType::ExcludeReads;
+                        },
+                        Some(EXCLUDE_WRITES) => {
+                            value  = ExcludeType::ExcludeWrites;
+                        },
+                        None => {},
+                        // Step 4.4 we dont return an error, just clear the result and break the loop.
+                        // The error handling can be found in the macro 'return_if_blacklisted'.(zakorgy)
+                        _ => {
+                            result.clear();
+                            break;
+                        },
+                    }
+                    // Step 4.5 we dont return an error here too.(zakorgy)
```

Comment:
I think it's step 4.4 and 4.5 because 4.4 says : "Otherwise, return an error and abort these steps." And this statement is true, when the UUID is not valid.